gems/aws-partitions/lib/aws-partitions/endpoint_provider.rb

```diff
@@ -80,9 +98,16 @@ def endpoint_for(region, service, sts_regional_endpoints)
           region = service_cfg.fetch("partitionEndpoint", region)
         end
 
+        # Check for s3 IAD regional behavior
+        s3_iad_regional = service == 's3' &&
+          region == 'us-east-1' &&
+          s3_us_east_1_regional_endpoint == 'regional'
+
```

Comment:
Is this variable necessary? In either case, wrap the boolean expression in parenthesis before assigning or checking `unless`.